File: app/src/main/java/com/yfbx/widgets/fragment/SoundWaveFrag.java
package com.yfbx.widgets.fragment;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.MotionEvent;
import android.view.View;

import com.yfbx.widgets.R;
import com.yfbx.widgets.java.anim.SoundWave;

/**
 * Author: Edward
 * Date: 2018/12/10
 * Description:
 */


public class SoundWaveFrag extends BaseFragment {

    private SoundWave soundWave;

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        soundWave = view.findViewById(R.id.sound_wave);
    }

    @Override
    public int getLayout() {
        return R.layout.frag_sound_wave;
    }

    public boolean onTouchEvent(MotionEvent event) {
        if (event.getAction() == MotionEvent.ACTION_UP) {
            soundWave.stop();
        } else {
            soundWave.start();
        }
        return true;
    }

}

File: app/src/main/java/com/yfbx/widgets/fragment/SoundWaveFrag.kt
package com.yfbx.widgets.fragment

import android.os.Bundle
import android.view.MotionEvent
import android.view.View

import com.yfbx.widgets.R
import com.yfbx.widgets.widget.anim.SoundWave

/**
 * Author: Edward
 * Date: 2018/12/10
 * Description:
 */


class SoundWaveFrag : BaseFragment() {

    private var soundWave: SoundWave? = null

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        soundWave = view.findViewById(R.id.sound_wave)
    }

    override fun getLayout(): Int {
        return R.layout.frag_sound_wave
    }

    fun onTouchEvent(event: MotionEvent): Boolean {
        if (event.action == MotionEvent.ACTION_UP) {
            soundWave!!.stop()
        } else {
            soundWave!!.start()
        }
        return true
    }

}
